### ISSUE TITLE
False positive for go.gorilla.security.audit.handler-assignment-from-multiple-sources.handler-assignment-from-multiple-sources ?

### ISSUE THREAD

**Issue by pm98zz-c** [returntocorp/sgrep-rules#1378]
More a "open for debate" than a proper bug. The rule chokes on multiple "error" assignments to the "err" variable.
Eg something like:

```
err := r.ParseForm()
if err != nil {
  log.Error(err)
}
var data Info
err = decoder.Decode(&data, r.PostForm)
if err != nil {
...
```

Now, arguably, the rule does its job and the variable is definitively assigned twice. But it feels like it is quite an idiomatic practice in go code everywhere. Maybe a bad practice, that said, and the solution might be simply "don't do that, then". 
I wonder if there's a way around it, though, maybe simply making an exception for the "err" variable?

**Comment by pm98zz-c** [comment]
Note: separate issue than https://github.com/returntocorp/semgrep-rules/issues/353, which I've hit too.

**Comment by minusworld** [comment]
@pm98zz-c Thanks for filing. Do you find this rule useful? We have been debating about removing it anyway. 😅

**Comment by pm98zz-c** [comment]
Yes, makes sense to me to remove it altogether. Appreciate what it is trying to do, but seems too fiddly to ever be reliable. 
It would need porting to all routers out there, plus I doubt anyone actually does this kind of checks directly in the handler (most would defer to a middleware, or abstract that in a GetUser function) so it feels doomed to produce mainly false positive.

**Comment by minusworld** [comment]
Removed in https://github.com/returntocorp/semgrep-rules/pull/1372.
